webdriver/navigation.py

```diff
@@ -1,109 +1,96 @@
-
-
 # TODO(ato): 7.1 Get
 
 
-def test_get_current_url_no_browsing_context(session, new_window):
+def test_get_current_url_no_browsing_context(session, create_window):
     # 7.2 step 1
-    session.window_handle = new_window
+    session.window_handle = create_window()
     session.close()
-    with pytest.raises(webdriver.NoSuchWindowException):
-        session.url = "about:blank"
+
+    result = session.send_raw_command("GET", "url")
+
+    wd_assert.error(result, "no such window")
 
 
 def test_get_current_url_alert_prompt(session):
     # 7.2 step 2
     session.url = alert_doc
-    with pytest.raises(webdriver.UnexpectedAlertOpenException):
-        session.url = "about:blank"
 
+    result = session.send_raw_command("GET", "url")
+
+    wd_assert.success(result, alert_doc)
 
 def test_get_current_url_matches_location(session):
     # 7.2 step 3
     url = session.execute_script("return window.location.href")
     assert session.url == url
 
-
-def test_get_current_url_payload(http, session):
+def test_get_current_url_payload(session):
     # 7.2 step 4-5
     session.start()
-    with http.get("/session/%s/url" % session.session_id) as resp:
-        assert resp.status == 200
-        body = json.load(resp)
-    assert "value" in body
-    assert isinstance(body["value"], types.StringTypes)
 
+    result = session.send_raw_command("GET", "url")
+
+    assert result.status == 200
+    assert isinstance(result.body["value"], basestring)
 
 def test_get_current_url_special_pages(session):
     session.url = "about:blank"
-    assert session.url == "about:blank"
 
+    result = session.send_raw_command("GET", "url")
+
+    wd_assert.success(result, "about:blank")
 
 # TODO(ato): This test requires modification to pass on Windows
 def test_get_current_url_file_protocol(session):
     # tests that the browsing context remains the same
     # when navigated privileged documents
     session.url = "file:///"
-    assert session.url == "file:///"
 
+    result = session.send_raw_command("GET", "url")
+
+    wd_assert.success(result, "file:///")
 
 # TODO(ato): Test for http:// and https:// protocols.
 # We need to expose a fixture for accessing
 # documents served by wptserve in order to test this.
 
+def test_get_current_url_after_modified_location(session):
+    session.execute_script("window.location.href = 'about:blank#wd_test_modification'")
 
-def test_get_current_url_malformed_url(session):
-    session.url = "foo"
-    assert session.url
+    result = session.send_raw_command("GET", "url")
 
+    wd_assert.success(result, "about:blank#wd_test_modification")
 
-def test_get_current_url_after_modified_location(session):
-    session.execute_script("window.location.href = 'about:blank'")
-    assert session.url == "about:blank"
+def test_get_current_url_nested_browsing_context(session, create_frame):
+    session.url = "about:blank#wd_from_within_frame"
 
+    session.send_command("POST", "frame", dict(id=create_frame()))
```

Comment:
So I guess this is meant to switch to the frame? You could probably do {"element-6066-11e4-a52e-4f735466cecf": create_frame()}`.